indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,1120 @@
+      }
+      catch (IOException ex) {
+        log.error(ex, "Could not fetch counters");
+      }
+      final JobID jobID = job.getJobID();
+
+      final Path jobDir = getJobPath(jobID, job.getWorkingDirectory());
+      final FileSystem fs = jobDir.getFileSystem(job.getConfiguration());
+      final RemoteIterator<LocatedFileStatus> it = fs.listFiles(jobDir, true);
+      final List<Path> goodPaths = new ArrayList<>();
+      while (it.hasNext()) {
+        final LocatedFileStatus locatedFileStatus = it.next();
+        if (locatedFileStatus.isFile()) {
+          final Path myPath = locatedFileStatus.getPath();
+          if (ConvertingOutputFormat.DATA_SUCCESS_KEY.equals(myPath.getName())) {
+            goodPaths.add(new Path(myPath.getParent(), ConvertingOutputFormat.DATA_FILE_KEY));
+          }
+        }
+      }
+      if (goodPaths.isEmpty()) {
+        log.warn("No good data found at [%s]", jobDir);
+        return null;
+      }
+      final List<DataSegment> returnList = ImmutableList.copyOf(
+          Lists.transform(
+              goodPaths, new Function<Path, DataSegment>()
+              {
+                @Nullable
+                @Override
+                public DataSegment apply(final Path input)
+                {
+                  try {
+                    if (!fs.exists(input)) {
+                      throw new ISE(
+                          String.format(
+                              "Somehow [%s] was found but [%s] is missing at [%s]",
+                              ConvertingOutputFormat.DATA_SUCCESS_KEY,
+                              ConvertingOutputFormat.DATA_FILE_KEY,
+                              jobDir
+                          )
+                      );
+                    }
+                  }
+                  catch (final IOException e) {
+                    throw Throwables.propagate(e);
+                  }
+                  try (final InputStream stream = fs.open(input)) {
+                    return HadoopDruidConverterConfig.jsonMapper.readValue(stream, DataSegment.class);
+                  }
+                  catch (final IOException e) {
+                    throw Throwables.propagate(e);
+                  }
+                }
+              }
+          )
+      );
+      if (returnList.size() == segments.size()) {
+        cleanup(job);
+        return returnList;
+      } else {
+        throw new ISE(
+            "Tasks reported success but result length did not match! Expected %d found %d at path [%s]",
+            segments.size(),
+            returnList.size(),
+            jobDir
+        );
+      }
+    }
+    catch (InterruptedException | ClassNotFoundException e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  public long getLoadedBytes()
+  {
+    return loadedBytes;
+  }
+
+  public long getWrittenBytes()
+  {
+    return writtenBytes;
+  }
+
+  private static void setupClassPath(Job job, JobConf jobConf)
```

Comment:
I'll be happy to set it, but tried to touch the existing hadoop stuff as little as possible.

If this PR is generally agreed upon and it proves pretty stable in our tests then I'll be happy to migrate existing hadoop stuff over to this PR's "framework" of doing things.